packages/commonwealth/server/routes/generateTokenIdea.ts

```diff
@@ -0,0 +1,192 @@
+import { AppError, ServerError, blobStorage } from '@hicommonwealth/core';
```

Comment:
This should be moved to `/libs/model`... we are incrementally deprecating the server folder in favor a strong typed libs.

Reply:
actually i couldn't find a way for streaming content from the newer API, plz lmk if that's possible and I will update.

Reply:
@Rotorsoft I updated the endpoint to be framework agnostic per our async discussion. Thanks for all the help.

Reply:
looks good, tried to randomize an idea, and when launching the community got this:

```gerprint":"INTERNAL_SERVER_ERROR-SequelizeDatabaseError-community.createCommunity"}
    err: {
      "type": "DatabaseError",
      "message": "column \"chain_node_id\" does not exist",
      "stack":
          Error
              at Query.run (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/dialects/postgres/query.js:76:25)
              at <anonymous> (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/sequelize.js:650:28)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at PostgresQueryInterface.insert (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/dialects/abstract/query-interface.js:795:21)
              at model.save (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/model.js:4154:35)
              at Function.create (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/model.js:2305:12)
              at <anonymous> (/Users/roger/Projects/common/commonwealth/libs/model/src/community/CreateCommunity.command.ts:136:9)
              at <anonymous> (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/sequelize.js:1197:18)
              at body (/Users/roger/Projects/common/commonwealth/libs/model/src/community/CreateCommunity.command.ts:109:7)
              at command (/Users/roger/Projects/common/commonwealth/libs/core/src/framework/command.ts:32:13)
      "name": "SequelizeDatabaseError",
      "parent": {
        "type": "DatabaseError",
        "message": "column \"chain_node_id\" does not exist",
        "stack":
            error: column "chain_node_id" does not exist
                at Parser.parseErrorMessage (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:514:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:545:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:495:3)
                at Readable.push (node:internal/streams/readable:375:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 114,
        "name": "error",
        "severity": "ERROR",
        "code": "42703",
        "position": "414",
        "file": "parse_relation.c",
        "line": "3651",
        "routine": "errorMissingColumn",
        "sql": "INSERT INTO \"Topics\" (\"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"group_ids\") VALUES (DEFAULT,$1,$2,$3,$4,$5,$6,$7,$8) RETURNING \"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"deleted_at\",\"order\",\"default_offchain_template\",\"channel_id\",\"group_ids\",\"telegram\",\"weighted_voting\",\"chain_node_id\",\"token_address\",\"token_symbol\",\"vote_weight_multiplier\";",
        "parameters": [
          "General",
          "General discussions",
          true,
          false,
          "dogeburger-dbgr-community",
          "2024-10-21 19:10:07.665 +00:00",
          "2024-10-21 19:10:07.665 +00:00",
          []
        ]
      },
      "original": {
        "type": "DatabaseError",
        "message": "column \"chain_node_id\" does not exist",
        "stack":
            error: column "chain_node_id" does not exist
                at Parser.parseErrorMessage (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/roger/Projects/common/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:514:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:545:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:495:3)
                at Readable.push (node:internal/streams/readable:375:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 114,
        "name": "error",
        "severity": "ERROR",
        "code": "42703",
        "position": "414",
        "file": "parse_relation.c",
        "line": "3651",
        "routine": "errorMissingColumn",
        "sql": "INSERT INTO \"Topics\" (\"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"group_ids\") VALUES (DEFAULT,$1,$2,$3,$4,$5,$6,$7,$8) RETURNING \"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"deleted_at\",\"order\",\"default_offchain_template\",\"channel_id\",\"group_ids\",\"telegram\",\"weighted_voting\",\"chain_node_id\",\"token_address\",\"token_symbol\",\"vote_weight_multiplier\";",
        "parameters": [
          "General",
          "General discussions",
          true,
          false,
          "dogeburger-dbgr-community",
          "2024-10-21 19:10:07.665 +00:00",
          "2024-10-21 19:10:07.665 +00:00",
          []
        ]
      },
      "sql": "INSERT INTO \"Topics\" (\"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"group_ids\") VALUES (DEFAULT,$1,$2,$3,$4,$5,$6,$7,$8) RETURNING \"id\",\"name\",\"description\",\"featured_in_sidebar\",\"featured_in_new_post\",\"community_id\",\"created_at\",\"updated_at\",\"deleted_at\",\"order\",\"default_offchain_template\",\"channel_id\",\"group_ids\",\"telegram\",\"weighted_voting\",\"chain_node_id\",\"token_address\",\"token_symbol\",\"vote_weight_multiplier\";",
      "parameters": [
        "General",
        "General discussions",
        true,
        false,
        "dogeburger-dbgr-community",
        "2024-10-21 19:10:07.665 +00:00",
        "2024-10-21 19:10:07.665 +00:00",
        []
      ]
    }

Reply:
I tried to repro but works fine for me, even tried to see if I was missing a migration but all were run. Not sure why its breaking on your end.

https://github.com/user-attachments/assets/3a64d032-b5f6-4992-beb0-35d14134accf

Reply:
ok, it worked after resetting my db! Nice job!

I noticed that the generated description always exceeds the 220 char limit, maybe it should be extended to 500 or something like that